lib/posix/posix_internal.h

```diff
@@ -15,35 +15,25 @@
  */
 #define PTHREAD_OBJ_MASK_INIT 0x80000000
 
-enum pthread_state {
-	/* The thread is running and detached. */
-	PTHREAD_DETACHED = PTHREAD_CREATE_DETACHED,
-	/* The thread is running and joinable. */
-	PTHREAD_JOINABLE = PTHREAD_CREATE_JOINABLE,
-	/* The thread structure is unallocated and available for reuse. */
-	PTHREAD_TERMINATED,
-	/* A joinable thread exited and its return code is available. */
-	PTHREAD_EXITED
-};
-
 struct posix_thread {
```

Comment:
Given the use of multiple spinlocks to protect various parts of this struct, I'd love to see comments indicating which lock covers which fields.

Reply:
alternatively, just use a single spinlock for all thread data; given that it looks like most state transitions involve queue manipulation anyhow, I'm not seeing how the per-thread lock will gain much performance.

Reply:
There is only 1 spinlock `lock`

Reply:
Seems like qid and the queue itself are covered by the global q_lock? That was where I got confused, seeing code using different locks in ways that looked like they were supposed to be protecting the same data.